.github/workflows/build.yml

```diff
@@ -0,0 +1,28 @@
+name: Build and deploy WCAG Primer
+
+on:
+- push
+
+jobs:
+  # A really basic check to see if this builds okay.
+  only-build:
+    runs-on: ubuntu-latest
+    if: github.ref != 'refs/heads/main' || github.ref != 'refs/heads/master'
+    steps:
+    - uses: actions/checkout@v2
+    - uses: ruby/setup-ruby@v1
+      with:
+        bundler-cache: true
+    - run: bundle install
+    - run: bundle exec middleman build --clean
+
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/master'
+    steps:
+    - uses: actions/checkout@v2
+    - uses: ruby/setup-ruby@v1
+      with:
+        bundler-cache: true
+    - run: bundle install
+    - run: bundle exec rake publish
```

Comment:
As I understand it, this rake task is going to update the `gh-pages` branch with the latest build of the site. By default, `actions/checkout` will only checkout the single commit for the workflow, which I think means that the `gh-pages` branch won't exist locally in the runner, which _might_ be an issue – in which case I think you'll need to [set `fetch-depth: 0` to checkout the entire repo with all of its branches](https://github.com/actions/checkout#Fetch-all-history-for-all-tags-and-branches). 

I think we'll also want to [set `user.name` and `user.email` in the Git config](https://github.com/actions/checkout#push-a-commit-using-the-built-in-token).

Reply:
Good catch - thank you.

I've set the `fetch-depth` to zero on the `deploy` workflow as that needs all the other branches to add to `gh-pages`.

 I don't think that the `build` workflow needs all the commits or branches, since we're testing whether this latest commit builds okay.

The `user.name` and `user.email` have been set to `github-actions[bot]`.